### ISSUE TITLE
Training time quite high

### ISSUE THREAD

**Issue by sandeep-krutrim** [McGill-NLP/llm2vec#96]
Hi,

I'm attempting to perform MNTP Lora training on Meta Llama 3, but the estimated training time is significantly high. Sequences are grouped in seq-len=512. 

`able parameters:
trainable params: 41,943,040 || all params: 7,546,867,712 || trainable%: 0.5558
Loading dataset from disk: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 369/369 [00:01<00:00, 275.82it/s]
Loading dataset from disk: 100%|███████████████████████████████████████████████████████████████████████████████████████████████| 41/41 [00:00<00:00, 315.72it/s]
/disk1/sandeep/miniconda3/envs/llm2vec_env/lib/python3.10/site-packages/transformers/utils/import_utils.py:533: FutureWarning: `is_torch_tpu_available` is deprecated and will be removed in 4.41.0. Please use the `is_torch_xla_available` instead.
  warnings.warn(
[INFO|trainer.py:2078] 2024-06-12 13:40:26,333 >> ***** Running training *****
[INFO|trainer.py:2079] 2024-06-12 13:40:26,333 >>   Num examples = 59,667,306
[INFO|trainer.py:2080] 2024-06-12 13:40:26,333 >>   Num Epochs = 3
[INFO|trainer.py:2081] 2024-06-12 13:40:26,333 >>   Instantaneous batch size per device = 4
[INFO|trainer.py:2083] 2024-06-12 13:40:26,333 >>   Training with DataParallel so batch size has been adjusted to: 32
[INFO|trainer.py:2084] 2024-06-12 13:40:26,333 >>   Total train batch size (w. parallel, distributed & accumulation) = 256
[INFO|trainer.py:2085] 2024-06-12 13:40:26,333 >>   Gradient Accumulation steps = 8
[INFO|trainer.py:2086] 2024-06-12 13:40:26,334 >>   Total optimization steps = 699,225
[INFO|trainer.py:2087] 2024-06-12 13:40:26,337 >>   Number of trainable parameters = 567,279,616
  0%|                                                                                                                                | 0/699225 [00:00<?, ?it/s][WARNING|logging.py:329] 2024-06-12 13:40:44,917 >> `use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`.
[WARNING|logging.py:329] 2024-06-12 13:40:44,918 >> `use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`.
/disk1/sandeep/miniconda3/envs/llm2vec_env/lib/python3.10/site-packages/torch/nn/parallel/_functions.py:68: UserWarning: Was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector.
  warnings.warn('Was asked to gather along dimension 0, but all '
  0%|                                                                                                                 | 6/699225 [10:56<21997:22:09, 113.26s  0%|             `

**Comment by vaibhavad** [comment]
Hi @sandeep-krutrim,

Are you sure the training is running on a GPU? Do you see GPU utilization statistics? Also, in our paper we stop the training after 1000 steps. Hence it does not go though the entire training dataset.

**Comment by vaibhavad** [comment]
Closing as it is stale. Feel free to re-open if you have any more questions.
